GOVERNANCE_DOC.md

```diff
@@ -0,0 +1,597 @@
+  OptionSet         []string            //  Options available to voters. {Yes, No, NoWithVeto, Abstain}
+  ProposalTypes     []string            //  Types available to submitters. {PlainTextProposal, SoftwareUpgradeProposal}
+  Threshold         uint64              //  Minimum value of Yes votes to No votes ratio for proposal to pass. Initial value: 0.5
+  Veto              rational.Rational   //  Minimum value of Veto votes to Total votes ratio for proposal to be vetoed. Initial value: 1/3
+  MaxDepositPeriod  uint64              //  Maximum period for Atom holders to deposit on a proposal. Initial value: 2 months
+  GovernancePenalty uint64              //  Penalty if validator does not vote
+  
+  ProcedureNumber   uint16              //  Incremented each time a new procedure is created
+  IsActive          bool                //  If true, procedure is active. Only one procedure can have isActive true.
+}
+```
+
+### Proposals
+
+`Proposals` are item to be voted on. They can be submitted by any Atom holder via a `TxSubmitProposal` transaction.
+
+```Go
+type TxSubmitProposal struct {
+  Title           string        //  Title of the proposal
+  Description     string        //  Description of the proposal
+  Type            string        //  Type of proposal. Initial set {PlainTextProposal, SoftwareUpgradeProposal}
+  Category        bool          //  false=regular, true=urgent
+  InitialDeposit  uint64        //  Initial deposit paid by sender. Must be strictly positive.
+}
+
+type Proposal struct {
+  Title             string              //  Title of the proposal
+  Description       string              //  Description of the proposal
+  Type              string              //  Type of proposal. Initial set {PlainTextProposal, SoftwareUpgradeProposal}
+  Category          bool                //  false=regular, true=urgent
+  Deposit           uint64              //  Current deposit on this proposal. Initial value is set at InitialDeposit
+  SubmitBlock       uint64              //  Height of the block where TxSubmitProposal was included
+  VotingStartBlock  uint64              //  Height of the block where MinDeposit was reached. -1 if MinDeposit is not reached.
+  Votes             map[string]uint64   //  Votes for each option (Yes, No, NoWithVeto, Abstain)
+}
+```
+
+Each `Proposal` is identified by its unique `proposalID`. 
+
+Additionaly, four lists will be linked to each proposal:
+- `DepositorList`: List of addresses that deposited on the proposal with their associated deposit
+- `VotersList`: List of addresses that voted **under each validator** with their associated option
+- `InitVotingPowerList`: Snapshot of validators' voting power **when proposal enters voting period** (only saves validators whose voting power is >0).
+- `MinusesList`: List of minuses for each validator. Used to compute validators' voting power when they cast a vote.
```

Comment:
We likely probably not be holding lists, but rather tallies of the voting power which has voted and in which way, and then have a unique item in the store for each item if they need to be referenced... we should never need to iterate through any of these lists :) - There are many parrellels to this architecture and the staking spec... In the staking spec we don't hold a list of the bonds to a validator, only the sum of the bonds in the validator, and then a bunch of unique bonds in the store, but no list of it all... neat aye, we should do something like that here

Reply:
Yeah initially I wanted mappings but AFAIK you can't map PubKey to base type in Go. I'm too unexperienced in Go to know the best way to circumvent that fact. All I want is to link `ProposalID` to a mapping where I can find an info by giving a PubKey. Don't need to iterate then. @sunnya97 probably knows better than me on this.

Reply:
sure, let's hear it sunny! this is an important implementation detail